csharp/tools/ZeroC.Ice.Slice.Tools/Common/SliceCompilerTask.cs

```diff
@@ -114,9 +114,8 @@ protected override int ExecuteTool(string pathToTool, string responseFileCommand
                 foreach (ITaskItem source in Sources)
                 {
                     var inputs = new List<string>();
-                    var depends = dependsDoc.DocumentElement.SelectNodes(
-                        string.Format("/dependencies/source[@name='{0}']/dependsOn",
-                                        source.GetMetadata("Identity")));
```

Comment:
This use XPath was brittle as it relies on string match, the new helper FindSourceNode uses FileInfo to compare file paths.